kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ServiceCrudTest.java

```diff
@@ -40,6 +43,14 @@ public void testCrud() {
     Service service2 = new ServiceBuilder().withNewMetadata().withName("svc2").addToLabels("foo", "bar").and().withNewSpec().and().build();
     Service service3 = new ServiceBuilder().withNewMetadata().withName("svc3").addToLabels("foo", "bar").and().withNewSpec().and().build();
 
+    // try to patch/replace before the service exists
+    KubernetesClientException result = assertThrows(KubernetesClientException.class,
+        () -> client.services().inNamespace("ns2").withName("svc2").patch(service1));
+    assertEquals(HttpURLConnection.HTTP_NOT_FOUND, result.getCode());
+    result = assertThrows(KubernetesClientException.class,
+        () -> client.services().inNamespace("ns2").withName("svc2").replace(service1));
+    assertEquals(HttpURLConnection.HTTP_NOT_FOUND, result.getCode());
```

Comment:
Using chained calls in assertThrows results in a code smell about not having single invocation throwing RuntimeException. Since we're doing it twice here, is it a good idea to refactor this like:
```java
    // try to patch/replace before the service exists
    Resource<Service> serviceOp = client.services().inNamespace("ns2").withName("svc2");
    KubernetesClientException result = assertThrows(KubernetesClientException.class,
        () -> serviceOp.patch(service1));
    assertEquals(HttpURLConnection.HTTP_NOT_FOUND, result.getCode());
    result = assertThrows(KubernetesClientException.class,
        () -> serviceOp.replace(service1));
    assertEquals(HttpURLConnection.HTTP_NOT_FOUND, result.getCode());
```

Reply:
The tests have been updated.

Reply:
Thanks, I've already approved your PR. There were just minor comments. Thanks for addressing them